### PR TITLE
bulk audit event export api

### PR DESCRIPTION
**Pull request by fspmarshall**
This PR implements a new api for exporting very large numbers of audit events in bulk very quickly.  This change is intended to support ongoing work related to https://github.com/gravitational/teleport/issues/46193.  The problem, in short, is that some teleport clusters have so much activity that our existing APIs for exporting events aren't performant enough.  In particular, reading audit events via Athena is very slow.  The new API implemented here bypasses Athena in favor of reading audit events directly from the parquet files in s3.  Bypassing Athena loses us the ability to exactly time-order events.  The export api is intended to prefer yielding events in time order, but makes no guarantees and may yield events out of order, especially in clusters that are under high load or when exporting historical data.

Use of the export api is a two-step process.  First, chunks of events for a given date are listed, then calls are made to stream the contents of each chunk, with per-chunk streaming being resumable via a cursor.  The api was designed this way for a few reasons: it maps well to the athena backend, it makes no assumptions about ordering abilities of the backend more granular than date, it makes it easy to poll past dates for changes in order to catch "straggler" events, and it provides an easy and powerful means of resumable parallelization since each chunk maintains an independent cursor state.

This PR also adds a hidden test utility to the `loadtest` subcommand for performing bulk export of events starting from a target date.

#### Performance

The per-chunk streaming API has a real-world throughput of over 200k events per minute, compared to a best-case 20k observed when querying through Athena (typically closer to 10-15k).  When listing chunks and streaming them synchronously, actual throughput drops to about 150k per minute if chunks are small and numerous.  The exact limits of export via parallel streaming of chunks haven't been determined.

### CODE REVIEW

File: lib/events/test/suite.go [hunk 1]
```diff
@@ -86,6 +88,128 @@ type EventsSuite struct {
...
+				eventCount++
+			}
+			require.NoError(t, events.Done())
+		}
+
+		require.NoError(t, chunks.Done())
+
+		require.Equal(t, 1, chunkCount)
+		require.Equal(t, 4, eventCount)
+	}, 120*time.Second, 500*time.Millisecond)
```
[the remark this answers is not in the document]
Comment (fspmarshall) [reply lib/events/test/suite.go]: accidental holdover from some debugging I was doing.  changed to 30s.

File: lib/events/athena/consumer.go [hunk 2]
```diff
@@ -161,7 +161,14 @@ func newConsumer(cfg Config, cancelFn context.CancelFunc) (*consumer, error) {
 		sqsDeleter:          sqsClient,
 		queueURL:            cfg.QueueURL,
 		perDateFileParquetWriter: func(ctx context.Context, date string) (io.WriteCloser, error) {
-			key := fmt.Sprintf("%s/%s/%s.parquet", cfg.locationS3Prefix, date, uuid.NewString())
+			// use uuidv7 to give approximate time order to files. this isn't strictly necessary
+			// but it assists in bulk event export roughly progressing in time order through a given
+			// day which is what folks tend to expect.
+			id, err := uuid.NewV7()
+			if err != nil {
+				return nil, trace.Wrap(err)
```
[the remark this answers is not in the document]
Comment (fspmarshall) [reply lib/events/athena/consumer.go:169]: IIUC if this callback fails the event will remain in the SQS queue and processing will be re-attempted later.

File: lib/events/multilog.go [hunk 3]
```diff
@@ -77,6 +80,62 @@ func (m *MultiLog) SearchEvents(ctx context.Context, req SearchEventsRequest) (e
...
+			foundImplemented = true
+			return err
+		}), nil
+	}))
+}
+
+func (m *MultiLog) GetEventExportChunks(ctx context.Context, req *auditlogpb.GetEventExportChunksRequest) stream.Stream[*auditlogpb.EventExportChunk] {
+	var foundImplemented bool
+	var pos int
+	// model our iteration through sub-loggers as a stream of streams that termiantes after the first stream
```
Comment (timothyb89) [inline lib/events/multilog.go:114]:
```suggestion
	// model our iteration through sub-loggers as a stream of streams that terminates after the first stream
```

**Review by timothyb89 — APPROVED** [review]
Seems pretty reasonable, just a few nits

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**